### PR TITLE
[grafana] Revert "Merge pull request #2977 from Sheikh-Abubaker/auto-mount-pr"

### LINKED ISSUE #2986: Grafana chart version `7.3.2` breaks k8s-sidecar
Discussed here 

https://github.com/grafana/helm-charts/pull/2977

### PR DESCRIPTION
**Pull request by Sheikh-Abubaker**
This reverts commit 49b699463f630bc454cf6b0ac3c55047031fb6e6, reversing changes made to d98e724868aa19e767537fe047abeb50a8283b93.

Fixes #2986

### CODE REVIEW

File: charts/grafana/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: grafana
-version: 7.3.2
+version: 7.3.1
```
Comment (onedr0p) [inline charts/grafana/Chart.yaml:3]:
```suggestion
version: 7.3.3
```

Comment (Sheikh-Abubaker) [reply charts/grafana/Chart.yaml]: done!!

**Review by onedr0p — APPROVED** [review]

**Review by Footur — APPROVED** [review]

**Comment by jkroepke** [comment]
Hi all,

instead fallback to an insecure default. Can we have a dedicated value for the pod.spec and set it to true?

https://github.com/grafana/helm-charts/blob/2f05c7b00057704e1d905ac8a88bc1e27e066e9f/charts/grafana/templates/_pod.tpl#L8

Running with serviceAccount.automountServiceAccountToken=false and pod.spec.automountServiceAccountToken=true (opt-in, where the SA is needed) is the general recommendation from security point of view.

**Comment by Sheikh-Abubaker** [comment]
@jkroepke

I believe your proposal is far more efficient and secure, it would be good if maintainers were to go forward with it

**Comment by zanhsieh** [comment]
Close in favour of #2991 .

**Pull request closed without merging**